modules/grpc/otel/filterx/object-otel-kvlist.cpp

```diff
@@ -170,6 +170,30 @@ KVList::has_subscript(FilterXObject *key) const
   return !!get_mutable_kv_for_key(key_c_str);
 }
 
+bool
+KVList::unset_key(FilterXObject *key)
+{
+  const gchar *key_c_str = filterx_string_get_value(key, NULL);
+  if (!key_c_str)
+    {
+      msg_error("FilterX: Failed to unset OTel KVList element",
+                evt_tag_str("error", "Key must be string type"));
+      return false;
+    }
+
+  for (int i = 0; i < repeated_kv->size(); i++)
+    {
+      KeyValue &possible_kv = repeated_kv->at(i);
+      if (possible_kv.key().compare(key_c_str) == 0)
+        {
+          repeated_kv->DeleteSubrange(i, 1);
+          return true;
+        }
+    }
+
+  return true;
```

Comment:
Why do we return true on this path? We couldn't unset anything.

Reply:
We concluded that unset should succeed unless there's an error